File: server/src/main/java/de/ketrwu/portfolio/controller/control/RedirectionController.java
package de.ketrwu.portfolio.controller.control;

import de.ketrwu.portfolio.controller.content.TextPageController;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;

@Controller
public class RedirectionController {

    @Autowired
    private TextPageController textPageController;

    @GetMapping({ "/impressum", "/imprint" })
    private String impressum() {
        return textPageController.getTextPage("imprint");
    }

    @GetMapping({ "/datenschutz", "/datenschutzerklaerung", "/datenschutzerklärung", "/privacy" })
    private String datenschutz() {
        return textPageController.getTextPage("privacy");
    }

}

File: server/src/main/kotlin/de/ketrwu/portfolio/controller/control/RedirectionController.kt
package de.ketrwu.portfolio.controller.control

import de.ketrwu.portfolio.controller.content.TextPageController
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping

/**
 * Controller that serves TextPages under another alias-path
 * @author Kenneth Wußmann
 */
@Controller
class RedirectionController {

    @Autowired
    private val textPageController: TextPageController? = null

    @GetMapping("/impressum", "/imprint")
    private fun impressum(model: MutableMap<String, Any>): String {
        return textPageController?.getTextPage("imprint", model) ?: "/"
    }

    @GetMapping("/datenschutz", "/datenschutzerklaerung", "/datenschutzerklärung", "/privacy")
    private fun datenschutz(model: MutableMap<String, Any>): String {
        return textPageController?.getTextPage("privacy", model) ?: "/"
    }
}
